modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/processors/ExternalRefProcessorTest.java

```diff
@@ -206,4 +211,19 @@ public void testRelativeRefIncludingUrlRef()
         is("./relative-with-url/relative-with-local.yaml#/relative-same-file")
     );
 	}
+
+    @Test
+    public void testHandleComposedSchemasInArrayItems() {
```

Comment:
would also test the case with `properties` alongside `oneOf` e.g. with following spec:

```
components:
  schemas:
    Response:
      type: array
      items:
        oneOf:
          - $ref: '#/components/schemas/ProductRow'
        properties: 
          orderRow:
            $ref: '#/components/schemas/OrderRow'
        discriminator:
          propertyName: type
          mapping:
            'product': '#/components/schemas/ProductRow'
    ProductRow:
      type: object
      additionalProperties: false
      required:
        - type
      properties:
        type:
          $ref: '#/components/schemas/ProductsRowType'
        payload:
          type: string
    ProductsRowType:
      type: string
      enum:
        - product    
    OrderRow:
      type: object
      additionalProperties: false
      required:
        - type
      properties:
        type:
          $ref: '#/components/schemas/OrderRowType'
        payload:
          type: string
    OrderRowType:
      type: string
      enum:
        - order

```

Reply:
Did not realise those were not mutually exclusive. Thank you for noticing. Fixed.